tests/integration/workflows/test_workflows_tasks_refextract.py

```diff
@@ -109,3 +112,166 @@ def test_match_reference_for_data_config():
+    """Test reference matcher for when inspire-matcher returns multiple matches
+    where the matched record id is one of the previous matched record id as well"""
+
+    original_cited_record_json = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'control_number': 1,
+        'document_type': ['article'],
+        'publication_info': [
+            {
+                'artid': '159',
+                'journal_title': 'JHEP',
+                'journal_volume': '03',
+                'page_start': '159',
+                'year': 2016
+            },
+            {
+                'artid': '074',
+                'journal_title': 'JHEP',
+                'journal_volume': '05',
+                'material': 'erratum',
+                'page_start': '074',
+                'year': 2017
+            }
+        ]
+    }
+
+    errata_cited_record_json = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'control_number': 2,
+        'document_type': ['article'],
+        'publication_info': [
+            {
+                'artid': '074',
+                'journal_title': 'JHEP',
+                'journal_volume': '05',
+                'material': 'erratum',
+                'page_start': '074',
+                'year': 2017
+            }
+        ]
+    }
+
+    TestRecordMetadata.create_from_kwargs(
+        json=original_cited_record_json, index_name='records-hep')
+
+    TestRecordMetadata.create_from_kwargs(
+        json=errata_cited_record_json, index_name='records-hep')
+
+    references = [
+        {
+            'reference': {
+                'publication_info': {
+                    'artid': '159',
+                    'journal_title': 'JHEP',
+                    'journal_volume': '03',
+                    'page_start': '159',
+                    'year': 2016
+                }
+            }
+        },
+        {
+            'reference': {
+                'publication_info': {
+                    'artid': '074',
+                    'journal_title': 'JHEP',
+                    'journal_volume': '05',
+                    'page_start': '074',
+                    'year': 2017
+                }
+            }
+        }
+    ]
+
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    assert validate(references, subschema) is None
+
+    matched_references = match_references(references)
+
+    assert matched_references[1]['record']['$ref'] == 'http://localhost:5000/api/literature/1'
+    assert validate(matched_references, subschema) is None
+
+
+def test_match_references_no_match_when_multiple_match_different_from_previous():
+    """Test reference matcher for when inspire-matcher returns multiple matches
+    where the matched record id is not the same as the previous matched record id"""
```

Comment:
Nit: you can just have a title such as `test_match_references_handles_conflicting_errata_citations` and then you don't have to explain what your test does in the docstring (which nobody reads, and is guaranteed to get stale with time.)